prow/cmd/sinker/main.go

```diff
@@ -105,17 +105,24 @@ func (c *controller) clean() {
 		c.logger.WithError(err).Error("Error listing prow jobs.")
 		return
 	}
+
+	// Only delete pod if its prowjob is marked as finished
+	isFinished := make(map[string]bool)
+
 	maxProwJobAge := c.configAgent.Config().Sinker.MaxProwJobAge
 	for _, prowJob := range prowJobs {
 		// Handle periodics separately.
 		if prowJob.Spec.Type == kube.PeriodicJob {
 			continue
 		}
-		if prowJob.Complete() && time.Since(prowJob.Status.StartTime) > maxProwJobAge {
-			if err := c.kc.DeleteProwJob(prowJob.Metadata.Name); err == nil {
-				c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Deleted prowjob.")
-			} else {
-				c.logger.WithFields(pjutil.ProwJobFields(&prowJob)).WithError(err).Error("Error deleting prowjob.")
+		if prowJob.Complete() {
```

Comment:
nit: can you structure this as
```go
if !prowJob.Complete() {
  continue
}
isFinished[prowJob.Metadata.Name] = true
if time.Since(prowJob.Status.StartTime) <= maxProwJobAge {
  continue
}
...
```
to avoid two levels of indentation?